src/main/java/org/dependencytrack/policy/PolicyEngine.java

```diff
@@ -84,8 +93,11 @@ private List<PolicyViolation> evaluate(final QueryManager qm, final List<Policy>
                 int policyConditionsViolated = 0;
                 for (final PolicyEvaluator evaluator : evaluators) {
                     evaluator.setQueryManager(qm);
-                    if (policyConditionViolations.addAll(evaluator.evaluate(policy, component))) {
-                        policyConditionsViolated++;
+                    final List<PolicyConditionViolation> policyConditionViolationsFromEvaluator = evaluator.evaluate(policy, component);
+                    if (!policyConditionViolationsFromEvaluator.isEmpty()) {
+                        policyConditionViolations.addAll(policyConditionViolationsFromEvaluator);
+                        policyConditionsViolated += (int) policyConditionViolationsFromEvaluator.stream()
+                                .map(pcv -> pcv.getPolicyCondition()).sorted(policyConditionComparator).distinct().count();
```

Comment:
It might be a better idea to execute `sorted` and `distinct` on a stream of primitive values (here: the condition's IDs), as `equals` is not explicitly defined for `PolicyCondition`. That also avoids the custom `Comparator`. How about something like this?

```java
policyConditionsViolated += (int) policyConditionViolationsFromEvaluator.stream()
        .map(PolicyConditionViolation::getPolicyCondition)
        .map(PolicyCondition::getId)
        .sorted(Long::compareTo)
        .distinct()
        .count();
```

Reply:
Can't believe I didn't see that... Guess it had been a long day.
Just wondering if we really want the double mapping and not just do that in a single line? Also, sorting long doesn't really need a comparator, so personally, I'd leave it empty...

```
policyConditionsViolated += (int) policyConditionViolationsFromEvaluator.stream()
        .map(pcv -> pcv.getPolicyCondition().getId())
        .sorted()
        .distinct()
        .count();
```
But if yours is the general coding-style used, I can gladly implement it like that. :-)

Reply:
Wasn't trying to enforce a specific coding style, I just like to give examples of what I mean 😅

Your suggestion sounds good.

Reply:
Don't worry, I'm not saying you did -- it's just that the contribution documentation says to use the 'existing style' and I hadn't come across streams yet, so I wanted to make sure if that was maybe the preferred style. :smile: 

New commit pushed.